lib/ipam/ipam.go

```diff
@@ -226,10 +227,22 @@ func (c ipamClient) determinePools(requestedPoolNets []net.IPNet, version int) (
 
 		// Make sure each requested pool exists
 		for _, rp := range requestedPoolNets {
-			if pool, ok := pm[rp.String()]; !ok {
+			pool, ok := pm[rp.String()]
+			if !ok {
 				// The requested pool doesn't exist.
 				return nil, fmt.Errorf("the given pool (%s) does not exist, or is not enabled", rp.IPNet.String())
-			} else {
+			} else if len(pool.Spec.NodeSelector) == 0 {
+				// No nodeSelector specified
+				enabledPools = append(enabledPools, pool)
+			} else if node, ok := nodeKV.Value.(*model.Node); !ok {
```

Comment:
I don't think this check here is useful.

We really have two cases:
- A nodeSelector is provided, in which case the node _needs_ to be passed.
- A nodeSelector is not provided, in which case we may want to allow the node to be nil (but not aware of a use-case for that). 

In practice, an IP allocation should always occur attached to a particular Calico node, so the node should never actually be nil. So, I think we can just drop this case, and return an error when we do a `Nodes().Get()` in the calling function.

Reply:
Might be good to just update this branch to return an error - essentially, it the IPPool has a node selector, then it is an error for the given nodeKV to be nil.